packages/toolpad-app/src/toolpad/Home.tsx

```diff
@@ -334,55 +511,119 @@ export default function Home() {
-                          app={app}
-                          activeDeployment={activeDeployment}
-                          onDelete={() => setDeletedApp(app)}
-                        />
-                      );
-                    })
-                  : 'No apps yet';
-              default:
-                return '';
-            }
-          })()}
-        </Box>
-      </Container>
+      <TabPanel index={0} value={tabIndex}>
+        <Container>
+          <CreateAppDialog open={createDialogOpen} onClose={() => setCreateDialogOpen(false)} />
+          <Toolbar variant={'regular'} disableGutters sx={{ justifyContent: 'space-between' }}>
+            <Button onClick={() => setCreateDialogOpen(true)}>Create New</Button>
+            <ToggleButtonGroup
+              value={viewMode}
+              exclusive
+              onChange={handleViewModeChange}
+              aria-label="view mode"
+            >
+              <ToggleButton
+                value="list"
+                aria-label="list view"
+                color={viewMode === 'list' ? 'primary' : undefined}
+              >
+                <ViewListIcon />
+              </ToggleButton>
+              <ToggleButton
+                value="grid"
+                aria-label="grid view"
+                color={viewMode === 'grid' ? 'primary' : undefined}
+              >
+                <GridViewIcon />
+              </ToggleButton>
+            </ToggleButtonGroup>
+          </Toolbar>
+          {viewMode === 'grid' ? (
+            <Box
+              sx={{
+                display: 'grid',
+                gridTemplateColumns: {
+                  lg: 'repeat(4, 1fr)',
+                  md: 'repeat(3, 1fr)',
+                  sm: 'repeat(2, fr)',
+                  xs: 'repeat(1, fr)',
+                },
+                gap: 2,
+              }}
+            >
+              {(() => {
+                switch (status) {
+                  case 'loading':
+                    return <AppCard />;
+                  case 'error':
+                    return <Alert severity="error">{(error as Error)?.message}</Alert>;
+                  case 'success':
+                    return apps.length > 0
+                      ? apps.map((app) => {
+                          const activeDeployment = activeDeploymentsByApp?.[app.id];
+                          return (
+                            <AppCard
+                              key={app.id}
+                              app={app}
+                              activeDeployment={activeDeployment}
+                              onDelete={() => setDeletedApp(app)}
+                            />
+                          );
+                        })
+                      : 'No apps yet';
+                  default:
+                    return <AppCard />;
+                }
+              })()}
+            </Box>
+          ) : null}
+          <Table aria-label="apps list" size="medium">
```

Comment:
Should the table be there even when in viewMode "grid"?

Reply:
Nope, fixed this in the refactor to `AppView`